src/services/BudgetService.ts

```diff
@@ -183,4 +188,65 @@ export class BudgetService {
       await QuarterBudgetModel.updateBudget(budget)
     }
   }
+
+  static getUncappedRoundedPercentage(value: number, total: number): number {
```

Comment:
I'd move this out from the BudgetService and use it as a helper function as it not strictly budget-related.